journal.py

```diff
@@ -51,36 +67,83 @@ def all(cls, session=None):
             session = DBSession
         return session.query(cls).order_by(cls.date.desc()).all()
 
+    @classmethod
+    def search(cls, id, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter_by(id=id).one()
+
+    @classmethod
+    def update(cls, id, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        entry = session.query(cls).filter_by(id=id).one()
+        entry.title = title
+        entry.text = text
+        return entry
+
 
 @view_config(route_name='home', renderer='templates/index.jinja2')
-def _view(request):
+def index_view(request):
     entries = Entry.all()
-    return {'entries':entries}
+    return {'entries': entries, 'add_markdown': add_markdown}
```

Comment:
if `add_markdown` were a method of the `Entry` class, then here you would not need to pass it to your renderer as a separate item.  You could simply pass the list of entries.